### PR TITLE
Implement a way to limit auto-format before save to specific directories

### PR DESCRIPTION
**Pull request by itkach**
A lot of existing code bases are not using automatic formatting so it is necessary to be able to enable it selectively.

### CODE REVIEW

**Comment by proofit404** [comment]
Does `.blacken` marker file something widely used in the black community?

**Comment by itkach** [comment]
Not that I know of, no. But maybe they will start using it now :)

**Comment by scop** [comment]
I think using the standard directory locals mechanism for selective enablement would be better than implementing new marker files. I guess this would be actually doable already now, with something conceptually like this in .dir-locals.el:

```elisp
((nil
  (eval . (add-hook 'python-mode-hook 'blacken-mode nil t))))
```
...except that it doesn't work :(. This however does work:
```elisp
((python-mode
  (eval . (add-hook 'before-save-hook 'blacken-buffer nil t))))
```
...but results in blacken-mode not actually turned on, thus suboptimal cooperation with that function, and lack of the "Black" modeline marker.

To summarize, if someone can come up with a snippet to .dir-locals.el that enables blacken-mode when python-mode is enabled in that dir, I think it would be a superior approach to defining a custom marker file. Another option would be to use a variable that would be set in .dir-locals.el for the opt-in instead of a custom separate marker file, but that's still worse than the add-hook completely in .dir-locals.el.

**Comment by proofit404** [comment]
Since it's not a widely used practice I don't want to invent one.
